### ISSUE TITLE
parser: `!:` annotations on variable declarators do not set `definite`

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#5999]
> Related to #5953 

For this snippet,
```ts
let x!: number = 1;
```
`VariableDeclarator::definite` should be `true` for `x` but it is not set. Likely because `!` is getting consumed when parsing type annotations.

Additionally, that snippet should produce a syntax error, but it does not.

* [Oxc Playground](https://playground.oxc.rs/#eNpFjsEOAiEMRH8Fe96LV4xX/wOwGBJoNwV0Nxv+XYhhPXVe28zMAQ40RCxqu2hFNVkUdVfXGyzAoA+QSmPknYrZQBepuEAMVKbOjlc8YU+W46QihrJnSaC9iRnbAquRjNIdux4uU5+vc+H4iS8c2R1SoODD/0RFOD4if0a1N4rl3Cv8Mlr7AoLmSbk=)
* [TypeScript Playground](https://www.typescriptlang.org/play/?ssl=1&ssc=20&pln=1&pc=1#code/DYUwLgBAHghAXBAdgVwLYCMQCcIF4ICMA3EA)
